PGM/src/main/java/tc/oc/pgm/map/MapDefinition.java

```diff
@@ -50,7 +51,8 @@
 
     private final MapFolder folder;
 
-    protected @Nullable MapModuleContext context;
+    protected @Nullable SoftReference<MapModuleContext> context;
```

Comment:
This should be `private` (Do other classes need to be able to directly modify this?).

Reply:
This was already previously protected, so that PGMMap can have access to it if required, even if it's currently not being used

Reply:
Could we use methods(i.e. getters and setters or other delegate methods) to work with the context, instead of making the SoftReference mutable?

Reply:
The soft reference has to be mutable. When the map isn't loaded the reference will be null. When a map has been forcibly unloaded because of memory constraints, the reference won't be null but the referenced context will